ws-client/src/client.rs

```diff
@@ -167,8 +172,19 @@ impl WsClient {
 				max_concurrent_requests,
 			)
 			.await;
+			shutdown_sender.close();
 		});
-		Ok(Self { to_back, request_timeout, error: Mutex::new(ErrorFromBack::Unread(err_rx)) })
+		Ok(Self {
+			to_back,
+			request_timeout,
+			error: Arc::new(Mutex::new(ErrorFromBack::Unread(err_rx))),
+			shutdown_receiver,
+		})
+	}
+
+	/// True if the backend has not shutdown
+	pub async fn is_connected(&self) -> bool {
+		!self.shutdown_receiver.is_closed()
```

Comment:
I suppose this would be called quite frequently w.r.t your use-case and using a Mutex/RwLock is not good idea.